src/ocean/meta/codegen/CTFE.d

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+
+    Utilities for common CTFE operations needed for code generation
+
+    Copyright:
+        Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.codegen.CTFE;
+
+import ocean.meta.types.Qualifiers;
+
+/*******************************************************************************
+
+    Converts integer to string
+
+    Params:
+        i = integer to convert
+
+*******************************************************************************/
+
+istring toString ( long i )
+{
+    istring digit="0123456789";
+
+    if (i==0)
+        return "0";
+
+    bool neg = false;
+
+    if (i<0)
+    {
+        neg = true;
+        i = -i;
+    }
+
+    istring res;
+
+    while (i>0)
+    {
+        res = digit[i % 10] ~ res;
+        i /= 10;
+    }
+
+    if (neg)
+        return '-'~res;
+    else
+        return res;
+}
+
+unittest
+{
+    static assert (toString(42) == "42");
+    static assert (toString(-42L) == "-42");
+    static assert (toString(31) == "31");
+    static assert (toString(-31) == "-31");
+    static assert (toString(14u) == "14");
+    static assert (toString(14L) == "14");
+    static assert (toString(14UL )== "14");
+}
+
+/*******************************************************************************
+
+    Utility to strip the part before the '.' in a string.
+
+    Params:
+        name = string to scan
+
+*******************************************************************************/
+
+public istring stripQualifiedPrefix ( istring name )
+{
+    ptrdiff_t idx = 0;
+
+    while (idx < name.length && name[idx] != '.')
+        ++idx;
+
+    if (idx == name.length)
+        return name;
+    else
+        return name[idx + 1 .. $];
+}
+
+unittest
+{
+    static assert (stripQualifiedPrefix("aa.bb") == "bb");
+    static assert (stripQualifiedPrefix("aabb") == "aabb");
```

Comment:
You rejected a PR that would allow it in a reasonable way so I have no other choice.